### PR TITLE
Activate Unit and Integration Tests in CI

### PR DESCRIPTION
**Pull request by macrogreg**
### Summary

This change adds all respective test projects that exist so far to the respective automation solutions.
The CI scripts will pick this up and the tests will become active as CI validation gates.

### Related work items

* https://github.com/temporalio/sdk-dotnet/issues/13
* https://github.com/temporalio/sdk-dotnet/issues/14

### CODE REVIEW

File: Src/Test/SDK/Common/Temporal.Serialization/public/TestUnnamedContainerPayloadConverter.cs [hunk 1]
```diff
@@ -33,6 +33,8 @@ public void Test_UnnamedContainerPayloadConverter_TrySerialize_Null()
         [Trait("Category", "Common")]
         public void Test_UnnamedContainerPayloadConverter_TrySerialize_Unnamed_SerializedDataBacked()
         {
+            //CompositePayloadConverter instance = new(new IPayloadConverter[] { new UnnamedContainerPayloadConverter(),
```
Comment (bergundy) [inline Src/Test/SDK/Common/Temporal.Serialization/public/TestUnnamedContainerPayloadConverter.cs:36]: Leftover?

**Review by bergundy — APPROVED** [review]
Saw a bunch of warnings to fix formatting.
Should those be errors instead and block PRs?

File: Src/Test/SDK/Common/Temporal.Serialization/public/TestUnnamedContainerPayloadConverter.cs (hunk 1 above)
Comment (macrogreg) [reply Src/Test/SDK/Common/Temporal.Serialization/public/TestUnnamedContainerPayloadConverter.cs:36]: We had a Zoom discussion on this with @mightyshazam. I left the comment in the code to remember that we need to add a test case for that invocation.

**Comment by macrogreg** [comment]
> Saw a bunch of warnings to fix formatting. Should those be errors instead and block PRs?

They should. However, they did not exist on Windows and they came out of nowhere in Linux. I suspect, something to do with line endings. I want to enable the CI without blocking on those issues, and due to their mysterious nature I an not sure if I can resolve them very fast, so prefer to do it in a later PR.

@mightyshazam you are on a Mac, right? Do you get them when you build locally? Any ideas about the cause?
Once they are resolved, we should discuss treating warnings as errors in the CI.

**Comment by mightyshazam** [comment]
I do get warnings on the files `Validate.cs` and `ExceptionExtensions.cs`. As @macrogreg stated, I'm fairly certain it is a whitespace issue. There should only be two warnings, but, because the files are shared across multiple projects, they get compiled multiple times. This gives the appearance of many issues instead of 2 lines of code.

File: Src/Test/SDK/Common/Temporal.Serialization/public/TestUnnamedContainerPayloadConverter.cs (hunk 1 above)
Comment (macrogreg) [reply Src/Test/SDK/Common/Temporal.Serialization/public/TestUnnamedContainerPayloadConverter.cs:36]: Addressed in https://github.com/temporalio/sdk-dotnet/pull/81

**Pull request merged**